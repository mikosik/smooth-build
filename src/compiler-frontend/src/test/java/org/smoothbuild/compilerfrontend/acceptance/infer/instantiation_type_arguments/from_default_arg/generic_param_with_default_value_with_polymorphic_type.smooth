A myId(A a) = a;
(B)->A myFunc(A a, (B)->A f = myId) = f;
myValue = myFunc(7);
