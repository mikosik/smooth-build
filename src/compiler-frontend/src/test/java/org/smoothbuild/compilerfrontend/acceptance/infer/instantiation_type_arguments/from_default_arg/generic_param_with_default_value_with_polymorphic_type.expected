A myId<A>(A a)
  = a<>;
(A)->A myFunc~f<A>
  = myId<A>;
(B)->A myFunc<A,B>(A a, (B)->A f = myFunc~f)
  = f<>;
(Int)->Int myValue<>
  = myFunc<Int, Int>(7,myFunc~f<Int>);
